src/main.ts

```diff
@@ -53,6 +56,14 @@ export default class TasksPlugin extends Plugin {
         this.registerEditorExtension(newLivePreviewExtension());
         this.registerEditorSuggest(new EditorSuggestor(this.app, getSettings()));
         new Commands({ plugin: this });
+
+        // Register the watcher for reminders.
+        this.app.workspace.onLayoutReady(async () => {
+            const tasks = this.getTasks();
+            if (this.taskNotification && tasks) {
```

Comment:
What would happen if the user opened a brand new vault that didn't have an tasks, and then during that session, added some tasks with reminders.

My guess is that they wouldn't get any reminders until they restarted Obsidian?

Best to remove the `&& tasks` clause, I feel.